cmd/options/options_test.go

```diff
@@ -82,3 +86,98 @@ func TestSetNodeNameOrDie(t *testing.T) {
 		}
 	}
 }
+
+func TestValidOrDie(t *testing.T) {
+	fooMonitorConfigMap := types.ProblemDaemonConfigPathMap{}
+	fooMonitorConfigMap["foo-monitor"] = &[]string{"config-a", "config-b"}
+
+	emptyMonitorConfigMap := types.ProblemDaemonConfigPathMap{}
+
+	for _, test := range []struct {
+		npdo        NodeProblemDetectorOptions
+		expectPanic bool
+	}{
+		// Verifies k8s exporter validation logic.
+		{
+			npdo: NodeProblemDetectorOptions{
+				MonitorConfigPaths: fooMonitorConfigMap,
+			},
+			expectPanic: false,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				ApiServerOverride:  "",
+				EnableK8sExporter:  false,
+				MonitorConfigPaths: fooMonitorConfigMap,
+			},
+			expectPanic: false,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				ApiServerOverride:  "",
+				EnableK8sExporter:  true,
+				MonitorConfigPaths: fooMonitorConfigMap,
+			},
+			expectPanic: false,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				ApiServerOverride:  "127.0.0.1",
+				EnableK8sExporter:  true,
+				MonitorConfigPaths: fooMonitorConfigMap,
+			},
+			expectPanic: false,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				ApiServerOverride:  ":foo",
+				EnableK8sExporter:  true,
+				MonitorConfigPaths: fooMonitorConfigMap,
+			},
+			expectPanic: true,
+		},
+		// Verifies monitor config validation logic.
+		{
+			npdo: NodeProblemDetectorOptions{
+				MonitorConfigPaths: fooMonitorConfigMap,
+			},
+			expectPanic: false,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				MonitorConfigPaths: emptyMonitorConfigMap,
+			},
+			expectPanic: true,
+		},
+		{
+			npdo:        NodeProblemDetectorOptions{},
+			expectPanic: true,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				SystemLogMonitorConfigPaths: []string{"config-a"},
+				MonitorConfigPaths:          fooMonitorConfigMap,
+			},
+			expectPanic: true,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				CustomPluginMonitorConfigPaths: []string{"config-a"},
+				MonitorConfigPaths:             fooMonitorConfigMap,
+			},
+			expectPanic: true,
+		},
+		{
+			npdo: NodeProblemDetectorOptions{
+				SystemLogMonitorConfigPaths: []string{"config-a"},
+			},
+			expectPanic: true,
```

Comment:
Whenever deprecated fields got used (i.e. `SystemLogMonitorConfigPaths`, `CustomPluginMonitorConfigPaths`), it will be considered as invalid, and `ValidOrDie()` will panic.

Reply:
Done.